### ISSUE TITLE
Error occurred when running CreateLMDBs.lua

### ISSUE THREAD

**Issue by worldweary** [eladhoffer/ImageNet-Training#13]
Hi, I'm a newbie to Torch.
I'm trying to make LMDB with the ImageNet images, but when I do `th CreateLMDBs.lua` I see this error.

```
(DataProvider)===>Generating filenames from path/media/cplab/9203d56a-6801-4a68-8538-7cdde9e54442/cplab/ImageNet/ILSVRC2014_CLSLOC/train/n03598930
*** Error in `/home/cplab/torch/distro/install/bin/luajit': free(): invalid next size (normal): 0x00007fd9247cd8f0 ***
Aborted (core dumped)
```

Or sometimes I see this error.

```
(DataProvider)===>Generating filenames from path/media/cplab/9203d56a-6801-4a68-8538-7cdde9e54442/cplab/ImageNet/ILSVRC2014_CLSLOC/train/n03598930
*** Error in `/home/cplab/torch/distro/install/bin/luajit': malloc(): memory corruption: 0x00007fa888bd9830 ***
Aborted (core dumped)
```

To be specific, when I did `strace th CreateLMDBs.lua` I got this.

```
write(1, "\33[0m(DataProvider)===>Generating"..., 156(DataProvider)===>Generating filenames from path/media/cplab/9203d56a-6801-4a68-8538-7cdde9e54442/cplab/ImageNet/ILSVRC2014_CLSLOC/train/n03598930
) = 156
openat(AT_FDCWD, "/media/cplab/9203d56a-6801-4a68-8538-7cdde9e54442/cplab/ImageNet/ILSVRC2014_CLSLOC/train/n03598930", O_RDONLY|O_NONBLOCK|O_DIRECTORY|O_CLOEXEC) = 25
getdents(25, /* 820 entries */, 32768)  = 32768
getdents(25, /* 482 entries */, 32768)  = 19280
getdents(25, /* 0 entries */, 32768)    = 0
close(25)                               = 0
getcwd("/home/cplab/torch/ImageNet-Training", 256) = 36
stat("/media/cplab/9203d56a-6801-4a68-8538-7cdde9e54442/cplab/ImageNet/ILSVRC2014_CLSLOC/train/n03598930", {st_mode=S_IFDIR|0775, st_size=73728, ...}) = 0
getcwd("/home/cplab/torch/ImageNet-Training", 256) = 36
stat("/media/cplab/9203d56a-6801-4a68-8538-7cdde9e54442/cplab/ImageNet/ILSVRC2014_CLSLOC/train", {st_mode=S_IFDIR|0775, st_size=40960, ...}) = 0
getcwd("/home/cplab/torch/ImageNet-Training", 256) = 36
stat("/media/cplab/9203d56a-6801-4a68-8538-7cdde9e54442/cplab/ImageNet/ILSVRC2014_CLSLOC/train/n03598930/n03598930_3917.JPEG", {st_mode=S_IFREG|0644, st_size=81667, ...}) = 0
getcwd("/home/cplab/torch/ImageNet-Training", 256) = 36
open("/dev/tty", O_RDWR|O_NOCTTY|O_NONBLOCK) = 25
writev(25, [{"*** Error in `", 14}, {"/home/cplab/torch/distro/install"..., 43}, {"': ", 3}, {"free(): invalid next size (norma"..., 34}, {": 0x", 4}, {"00007f5f366af570", 16}, {" ***\n", 5}], 7*** Error in `/home/cplab/torch/distro/install/bin/luajit': free(): invalid next size (normal): 0x00007f5f366af570 ***
) = 119
mmap(NULL, 4096, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x7f5f587ab000
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
tgkill(13051, 13051, SIGABRT)           = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=13051, si_uid=1000} ---
rt_sigprocmask(SIG_UNBLOCK, [ABRT], NULL, 8) = 0
tgkill(13051, 13051, SIGABRT)           = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=13051, si_uid=1000} ---
rt_sigreturn()                          = 0
rt_sigaction(SIGABRT, {SIG_DFL, ~[], SA_RESTORER, 0x7f5f57871d40}, NULL, 8) = 0
tgkill(13051, 13051, SIGABRT)           = 0
--- SIGABRT {si_signo=SIGABRT, si_code=SI_TKILL, si_pid=13051, si_uid=1000} ---
+++ killed by SIGABRT (core dumped) +++
Aborted (core dumped)
```

I'd be glad if somebody could help me.

**Comment by worldweary** [comment]
I tried again with updated lua files, and the error disappeared! Thanks a lot!
But this time I got another error below.

```
(Container)===>Shuffling Items
Error in LMDB function mdb_cursor_get :         MDB_NOTFOUND: No matching key/data pair found
/home/cplab/torch/distro/install/bin/luajit: CreateLMDBs.lua:12: attempt to concatenate field 'ImageMinSide' (a nil value)
stack traceback:
        CreateLMDBs.lua:12: in function 'PreProcess'
        CreateLMDBs.lua:34: in function 'LoadImgData'
        CreateLMDBs.lua:61: in function 'LMDBFromFilenames'
        CreateLMDBs.lua:107: in main chunk
        [C]: in function 'dofile'
        ...rch/distro/install/lib/luarocks/rocks/trepl/scm-1/bin/th:131: in main chunk
        [C]: at 0x00406670
```

How can I fix it?

**Comment by yeqinglee** [comment]
I have encounter the same problem while running "CreateLMDBs.lua".
Probably it is due to the "SubFolders" option of ValidationFiles is set to false by default.
When I set it to true, problem solved (the validation set of ILSVR2012 is extracted to multiple sub-folders similar to the training set).
